receiver.go

```diff
@@ -569,20 +583,33 @@ func (r *Receiver) mux() {
 		})
 	}()
 
+	if r.autoSendFlow {
+		r.l.doneErr = r.muxFlow(r.l.linkCredit, false)
+	}
+
 	for {
 		msgLen := r.messagesQ.Len()
 
-		// max - (linkCredit + countUnsettled) == pending credit (i.e. credit we can reclaim)
+		r.settlementCountMu.Lock()
+		// counter that accumulates the settled delivery count.
+		// once the threshold has been reached, the counter is
+		// reset and a flow frame is sent.
+		settlementCount := r.settlementCount
```

Comment:
NIT: I found it helpful to rename this to 'previousSettlementCount', just to make it a bit easier to distinguish it from r.settlementCount.

Reply:
Fixed.